### PR TITLE
[7.4-only] LPS-122449 Replace usages of dom.delegate in Journal, Knowledge Base and Wiki

### PR DESCRIPTION
**Pull request by kresimir-coko**
This PR removes usages of `dom.delegate` from files found inside the `Journal`, `Knowledge Base` and `Wiki` modules and replaces those usages with the modern `delegate` utility.

Continuation of #531

### CODE REVIEW

**Comment by kresimir-coko** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:sf

**Review by wincent — APPROVED** [review]
lg2m

cc @liferay-lima (and more specifically @ambrinchaudhary and @boton)

Will forward if CI is green; but please feel free to jump in and `ci:close` the forwarded pull if you see any issues with this.

**Comment by wincent** [comment]
ci:forward

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/item/selector/select_articles.jsp [hunk 1]
```diff
@@ -373,8 +373,8 @@ JournalArticleItemSelectorViewDisplayContext journalArticleItemSelectorViewDispl
 	</liferay-ui:search-container>
 </clay:container-fluid>
 
-<aui:script require="metal-dom/src/all/dom as dom" sandbox="<%= true %>">
-	var selectArticleHandler = dom.delegate(
+<aui:script require="frontend-js-web/liferay/delegate/delegate.es as delegateModule" sandbox="<%= true %>">
+	var selectArticleHandler = delegate(
```
Comment (ambrinchaudhary) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/item/selector/select_articles.jsp:377]: Missing delegate variable declaration, isn't it? 
`var delegate = delegateModule.default;`

Comment (kresimir-coko) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/item/selector/select_articles.jsp:377]: It seems to be the case, yes @ambrinchaudhary I'll get on top of that ASAP

Comment (kresimir-coko) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/item/selector/select_articles.jsp:377]: @wincent what can we do to make sure the wrong version doesn't reach Brian?

File: modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/common/history.jsp [hunk 2]
```diff
@@ -183,7 +183,7 @@ if (portletTitleBasedNavigation) {
 	</liferay-ui:search-container>
 </aui:fieldset>
 
-<aui:script require="metal-dom/src/dom as dom">
+<aui:script require="metal-dom/src/dom as dom,frontend-js-web/liferay/delegate/delegate.es as delegateModule">
```
Comment (ambrinchaudhary) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/resources/META-INF/resources/admin/common/history.jsp:186]: I think this metal-dom/src/dom dependency cant be removed, I don't see its use.

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/item/selector/select_articles.jsp (hunk 1 above)
Comment (ambrinchaudhary) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/item/selector/select_articles.jsp:377]: It has already been forwarded, I'll close https://github.com/brianchandotcom/liferay-portal/pull/96359 and forward it again once this is fixed ;-)

Comment (kresimir-coko) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/item/selector/select_articles.jsp:377]: Thank you @ambrinchaudhary

**Comment by ambrinchaudhary** [comment]
ci:open

**Comment by ambrinchaudhary** [comment]
ci:reopen

**Comment by kresimir-coko** [comment]
@ambrinchaudhary I've pushed a commit that addresses your comments 😄

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/item/selector/select_articles.jsp (hunk 1 above)
Comment (wincent) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/item/selector/select_articles.jsp:379]: > Missing delegate variable declaration, isn't it?

I am glad somebody is awake!

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
> In software development, Linus's law is the assertion that "given enough eyeballs, all bugs are shallow".

https://en.wikipedia.org/wiki/Linus%27s_law

grinning

**Comment by wincent** [comment]
ci:forward

**Pull request closed without merging**